### DISCUSSION TITLE
I would like to draw the window borders in macOS

### DISCUSSION THREAD

**Discussion by josepheiba** [wezterm/wezterm#4936] [Q&A]
### What Operating System(s) are you running on?

macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

macOS window manager, I would like to draw the window borders in macOS, I tried using border_left_width top right bottom, but the corners wasn't rounded, and I don't think that's the solution, anyone have and idea how?
<img width="512" alt="Screenshot 2024-02-02 at 5 49 23 AM" src="https://github.com/wez/wezterm/assets/101673502/9d5a1926-84ab-4c79-b9a4-415e68541ae6">

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

I'm experiencing an issue with drawing minimal borders on macOS in WezTerm. I've tried using border_left_width..., but the borders are not rendering as rounded

**Comment by Faetu** [comment]
This is one reason why I always go back to iTerm2.
Wezterm is nice but dont look native because of the borders. I think we are doing something wrong, becaus the screenshot from the repo main readme, shows native macos borders: 
![](https://raw.githubusercontent.com/wez/wezterm/main/docs/screenshots/two.png)


@wez how can we achieve the same look, as you?

**Comment by Faetu** [comment]
Hi @wez

Anything new about this topic?

**Reply by naikrovek** [reply]
Wez said this in another request for update that you made, but I just want to state why I consider requests for updates to be rude: Maintainers know where to go when they have an update they wish to share.  Maintainers do not forget where to do this, or to do it at all.  

When someone asks for an update repeatedly, it very much feels like the requester is attempting to push the person into action; to drop anything they're already doing and to work on what the requester wants them to work on.  ⇦ this is why many consider it rude to ask for updates.

Please remember that this software is free and provided with no support.  Any support that you get is done out of kindness, a kindness which is often very tenuous and fragile.  Avoid doing anything which puts strain on the maintainer of software that you like and use; you could be the person that pushes the maintainer past the point where working on the software is rewarding for them.

I'm sure there is a cultural thing or English as a second language thing going on here, so please don't feel that I am picking on you or calling you out specifically.  I wish to address the behavior, and not the person.

If you do happen to be from a culture where regularly asking for status updates is a normal thing to do, please do what you can to change that culture.  Us westerners do not like being asked the same question by the same person repeatedly.

My intent with this comment is to be constructive and helpful, as well as explanatory.  I have no other intent with this comment.  If anyone views this as an attack, then I have failed in communicating, because that is not my intent as I type this, nor is it a feeling that I have as I type this.

**Comment by Faetu** [comment]
Hi @wez 

Some news here?

best regards

**Reply by wez** [reply]
Please stop pinging me. If/when I look into this, I will comment.

**Comment by mv-stns** [comment]
@Faetu incredibly rude of you.

**Comment by walkerk1980** [comment]
The borders don't exactly draw rounded aside from the scrollbar side, but it does prevent the edges from blending in with the background.

```lua
config.window_decorations = "INTEGRATED_BUTTONS|RESIZE" -- place window management buttons (minimize, maximize, close) into the tab bar instead of showing a title bar.

config.enable_scroll_bar = true

config.window_frame = {
  border_left_width = '1px',
  border_right_width = '1px',
  border_bottom_height = '1px',
  border_top_height = '0px',
  border_left_color = '151515',
  border_right_color = '151515',
  border_bottom_color = '151515',
  border_top_color = '151520',
}
```

![Screenshot 2024-12-31 at 2 01 48 PM](https://github.com/user-attachments/assets/7465193c-8a9f-4c78-9af8-6d8f239e2d33)

**Reply by Kamyil** [reply]
Exactly! I've only slightly changed the colors and I think it kinda looks like native
```lua
config.window_frame = {
	border_left_width = '1px',
	border_right_width = '1px',
	border_bottom_height = '1px',
	border_top_height = '1px',
	border_left_color = '#484848',
	border_right_color = '#484848',
	border_bottom_color = '#484848',
	border_top_color = '#484848',
}
```

**Comment by Kamyil** [comment]
It's quite easy to achieve with a config

```lua
config.window_frame = {
	border_left_width = '1px',
	border_right_width = '1px',
	border_bottom_height = '1px',
	border_top_height = '1px',
	border_left_color = '#484848',
	border_right_color = '#484848',
	border_bottom_color = '#484848',
	border_top_color = '#484848',
}
```

ofc. edges are a little bit rough here, but I think it's more than enough for me :)

<img width="3690" height="2044" alt="image" src="https://github.com/user-attachments/assets/24374ac3-6406-4fb3-8107-20d142192762" />
